pytest_socket.py

```diff
@@ -67,6 +69,18 @@ def socket_enabled(pytestconfig):
     yield
 
 
+@dataclass
+class _PytestSocketConfig:
+    socket_disabled: bool
+    socket_force_enabled: bool
+    allow_unix_socket: bool
+    allow_hosts: typing.Union[str, typing.List[str], None]
+    resolution_cache: typing.Dict[str, typing.Set[str]]
```

Comment:
I _think_ this can have a `default_factory` of an empty dict, so that when creating it later you don't have to pass an empty dict.

Refs: https://docs.python.org/3/library/dataclasses.html#dataclasses.field